stacksCollectible/index.ts

```diff
@@ -0,0 +1,145 @@
+
+  // get NFT collection data and NFT metadata in batches
+  const batchSize = 10;
+  for (let index = 0; index < nfts.length; index += batchSize) {
+    const nftDetailPromises: Array<Promise<NftDetailResponse>> = [];
+    const collectionDataPromises: Array<Promise<NftCollectionData | undefined>> = [];
+    const nftArray: Array<NonFungibleToken> = [];
+
+    const batch = nfts.slice(index, index + batchSize);
+    batch.forEach((nft) => {
+      const principal: string[] = nft.asset_identifier.split('::');
+      const contractInfo: string[] = principal[0].split('.');
+      const contractId = principal[0];
+
+      if (contractInfo[1] === 'bns') {
+        //no further data required for BNS, arrange into collection
+        const bnsCollection: StacksCollectionData = {
+          collection_id: 'bns',
+          collection_name: 'BNS Names',
+          total_nft: 1,
+          thumbnail_nfts: [nft],
+          all_nfts: [nft],
+        };
+        collectionRecord[contractInfo[1]] = bnsCollection;
+      } else {
+        nftDetailPromises.push(getNftDetail(nft.value.repr.replace('u', ''), contractInfo[0], contractInfo[1]));
+        collectionDataPromises.push(getNftsCollectionData(contractId));
+        nftArray.push(nft);
+      }
+    });
+    const resolvedNftDetailPromises = await Promise.all(nftDetailPromises);
+    const resolvedCollectionDataPromises = await Promise.all(collectionDataPromises);
+
+    for (let i = 0; i < nftArray.length; i++) {
+      const nft = nftArray[i];
+      const principal: string[] = nft.asset_identifier.split('::');
+      const contractInfo: string[] = principal[0].split('.');
+      const contractId = principal[0];
+
+      if (contractInfo[1] === 'bns') continue; //collection already assigned
+
+      const nftDetail = resolvedNftDetailPromises[i];
+      const collectionData = resolvedCollectionDataPromises[i];
+
+      if (nftDetail) nft.data = nftDetail.data;
+
+      //group NFTs into collections
+      if (collectionRecord[contractId]) {
+        const data = collectionRecord[contractId];
+
+        data.all_nfts.push(nft);
+        if (data.total_nft < 4) {
+          data?.thumbnail_nfts.push(nft);
+        }
+        data.total_nft += 1;
+      } else {
+        collectionRecord[contractId] = {
+          collection_id: contractId,
+          collection_name: collectionData?.collection?.name ?? contractId,
+          total_nft: 1,
+          all_nfts: [nft],
+          thumbnail_nfts: [nft],
+          floor_price: collectionData?.collection?.floorItem?.price
+            ? microstacksToStx(new BigNumber(collectionData?.collection?.floorItem?.price)).toNumber()
+            : 0,
+        };
+      }
+    }
+  }
+
+  const nftCollectionList = Object.values(collectionRecord);
+
+  //sort according to total nft in a collection
+  nftCollectionList.sort((a, b) => {
+    //place bns collection at the bottom of nft list
+    if (a.collection_id === 'bns' || b.collection_id === 'bns') return -1;
```

Comment:
You'll need to return 1 if a is bns and -1 if b is bns (or the other way around 🤔 )

Reply:
we basically want to prioritize other collections over the 'bns' collection when sorting. So with the current implementation, if either a or b is a bns collection we assign -1 so it always comes at the end of the list. From my testing, all bns names do come at the end of the collection list. Is there some other reason why this lines needs to be updated 🤔

Reply:
@Imamah-Zafar the way the sort function works is it iterates through the original array by looking at two adjacent items (assigns them a and b). your compareFn supplied to sort() will tell it whether to swap a and b around or not.

compareFn should return 1 for swap a and b, or return -1 for do not swap a and b.
it means line 133 should return either 1 or -1 depending on which item was the bns item

ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort
![image](https://github.com/secretkeylabs/xverse-core/assets/6109710/350ee296-a970-4f28-8e81-423e2a7d914e)

Reply:
oh i see. thanks for this. I have updated the sorting